src/main/java/com/amazon/sqs/javamessaging/SQSSession.java

```diff
@@ -731,114 +687,149 @@ void waitForCallbackComplete() {
+        throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
+    }
+
+    /**
+     * This method is not supported.
+     */
     @Override
     public QueueBrowser createBrowser(Queue queue) throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public QueueBrowser createBrowser(Queue queue, String messageSelector) throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public TemporaryQueue createTemporaryQueue() throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public TemporaryTopic createTemporaryTopic() throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public MessageListener getMessageListener() throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public void setMessageListener(MessageListener listener) throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public StreamMessage createStreamMessage() throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    /** This method is not supported. */
+    /**
+     * This method is not supported.
+     */
     @Override
     public MapMessage createMapMessage() throws JMSException {
         throw new JMSException(SQSMessagingClientConstants.UNSUPPORTED_METHOD);
     }
 
-    static class CallbackEntry {
-        private final MessageListener messageListener;
-        private final MessageManager messageManager;
-        
-        CallbackEntry(MessageListener messageListener, MessageManager messageManager) {
-            this.messageListener = messageListener;
-            this.messageManager = messageManager;
-        }
-        
-        public MessageListener getMessageListener() {
-            return messageListener;
-        }
-        
-        public MessageManager getMessageManager() {
-            return messageManager;
-        }
+    record CallbackEntry(MessageListener messageListener, MessageManager messageManager) {
```

Comment:
what is this method doing?

Reply:
I don't get it what you mean. The
```java
record CallbackEntry(MessageListener messageListener, MessageManager messageManager) {
}
```
record class is the replacement of [SQSSession.java](https://github.com/awslabs/amazon-sqs-java-messaging-lib/blob/master/src/main/java/com/amazon/sqs/javamessaging/SQSSession.java#L824-L840)